pkg/cmd/server/origin/master.go

```diff
@@ -131,249 +134,210 @@ import (
 )
 
 const (
-	LegacyOpenShiftAPIPrefix = "/osapi" // TODO: make configurable
-	OpenShiftAPIPrefix       = "/oapi"  // TODO: make configurable
-	KubernetesAPIPrefix      = "/api"   // TODO: make configurable
-	KubernetesAPIGroupPrefix = "/apis"  // TODO: make configurable
-	OpenShiftAPIV1           = "v1"
-	OpenShiftAPIPrefixV1     = OpenShiftAPIPrefix + "/" + OpenShiftAPIV1
-	swaggerAPIPrefix         = "/swaggerapi/"
+	openAPIServePath = "/swagger.json"
 	// Discovery endpoint for OAuth 2.0 Authorization Server Metadata
 	// See IETF Draft:
 	// https://tools.ietf.org/html/draft-ietf-oauth-discovery-04#section-2
 	oauthMetadataEndpoint = "/.well-known/oauth-authorization-server"
 )
 
-var (
-	excludedV1Types = sets.NewString()
+var excludedV1Types = sets.NewString()
 
-	// TODO: correctly solve identifying requests by type
-	longRunningRE = regexp.MustCompile("watch|proxy|logs?|exec|portforward|attach")
-)
+// Run launches the OpenShift master by creating a kubernetes master, installing
+// OpenShift APIs into it and then running it.
+func (c *MasterConfig) Run(kc *kubernetes.MasterConfig, assetConfig *AssetConfig) {
```

Comment:
this rewrite (and special-casing RunInProxyMode) makes it really hard to know what is protected and what isn't